api/agentpb/agent.go

```diff
@@ -81,10 +81,13 @@ func GetAgentConnectMetadata(ctx context.Context) AgentConnectMetadata {
 	return res
 }
 
-// AddAgentServerMetadata adds metadata to pmm-managed's Connect RPC call.
+// AddAgentServerMetadata adds metadata to pmm-managed's Connect RPC call and returns it for sending.
```

Comment:
It doesn't _adds_ anything. Either comment and method name should be updated, or it should become `SendAddAgentServerMetadata` and actually send it. I think the latter variant is better – that way caller doesn't have to think about headers vs trailers distinction.